extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -167,8 +183,20 @@ public SecurityIdentity apply(Throwable throwable) {
             }
         }
 
-        challenge = new ChallengeData(HttpResponseStatus.FOUND.code(), HttpHeaders.LOCATION,
-                configContext.auth.authorizeURL(params));
+        if (shouldAutoRedirect(configContext, context)) {
+            challenge = new ChallengeData(HttpResponseStatus.FOUND.code(), HttpHeaders.LOCATION,
+                    configContext.auth.authorizeURL(params));
+        } else {
+            // If the client (usually an SPA) wants to handle the redirect manually, then
+            // return status code 499 and in the www-authenticate header put the redirect url
+            // www-authenticate header content; No standard content matched the exact use case
+            // so the OAuth redirect-url= is an arbitrary string that can be tested
+            // for on the client side
+            String headerContent = "OIDC redirect-url=" + configContext.auth.authorizeURL(params);
+            // Vertx HTTPHeaders does noy have a "WWW_AUTHENTICATE" static
```

Comment:
You can drop this comment, it is OK if you just set `WWW-Authenticate`

Reply:
Done